presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -1368,6 +1369,106 @@ public static double greatCircleDistance(
         return atan2(sqrt(t1 * t1 + t2 * t2), t3) * EARTH_RADIUS_KM;
     }
 
+    @Description("Decodes a polyline to a linestring")
+    @ScalarFunction("decode_polyline")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice decodePolyline(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(decodePolyline(input.toStringUtf8()));
+    }
+
+    private static OGCLineString decodePolyline(String polyline)
+    {
+        int len = polyline.length();
+
+        MultiPath multipath = new Polyline();
+        boolean isFirstPoint = true;
+
+        int index = 0;
+        int lat = 0;
+        int lng = 0;
+
+        while (index < len) {
+            int result = 1;
+            int shift = 0;
+            int b;
+            do {
+                b = polyline.charAt(index++) - 63 - 1;
+                result += b << shift;
+                shift += 5;
+            } while (b >= 0x1f);
+            lat += (result & 1) != 0 ? ~(result >> 1) : (result >> 1);
+
+            result = 1;
+            shift = 0;
+            do {
+                b = polyline.charAt(index++) - 63 - 1;
+                result += b << shift;
+                shift += 5;
+            } while (b >= 0x1f);
+            lng += (result & 1) != 0 ? ~(result >> 1) : (result >> 1);
+
+            if (isFirstPoint) {
+                multipath.startPath(lng * 1e-5, lat * 1e-5);
+                isFirstPoint = false;
+            }
+            else {
+                multipath.lineTo(lng * 1e-5, lat * 1e-5);
+            }
+        }
+
+        return new OGCLineString(multipath, 0, null);
+    }
+
+    @Description("Encodes a linestring or multilinestring to a polyline")
+    @ScalarFunction("encode_polyline")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice encodePolyline(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        validateType("encode_polyline", geometry, EnumSet.of(LINE_STRING, MULTI_LINE_STRING));
+        return utf8Slice(encodePolyline(geometry));
+    }
+
+    private static String encodePolyline(OGCGeometry geometry)
+    {
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+
+        long lastLat = 0;
+        long lastLng = 0;
+
+        final StringBuilder result = new StringBuilder();
```

Comment:
We generally don't use `final` for local variables.

Reply:
This could encode directly to a Slice instance. You can use `DynamicSliceOutput`, append the values to it and then call `slice()` to get the built object.